homeassistant/config_entries.py

```diff
@@ -1614,9 +1614,11 @@ def _index_entry(self, entry: ConfigEntry) -> None:
             unique_id_hash = entry.unique_id
             if not isinstance(entry.unique_id, str):
                 # Guard against integrations using unhashable unique_id
-                # In HA Core 2024.9, we should remove the guard and instead fail
                 if not isinstance(entry.unique_id, Hashable):  # type: ignore[unreachable]
```

Comment:
I think these `if` statements can now be combined

Reply:
`if type(entry.unique_id) is not str and not isinstance(entry.unique_id, Hashable):` would likely be the fastest since the type check will be super fast and only fallback to Hashable (which is a bit expensive because it has to examine the object for `__hash__`)

Not sure how often its called though to know if its worth guarding with a fast type check

Reply:
But then we would also raise on non-string unique IDs, which was added in 2024.10

Reply:
Sorry about that. Looked on my phone and missed the indent